src/sentry/static/sentry/app/components/acl/feature.tsx

```diff
@@ -11,84 +11,96 @@ import withProject from 'app/utils/withProject';
 
 import ComingSoon from './comingSoon';
 
-type FeatureProps = {
+type Props = {
+  /**
+   * The following properties will be set by the HoCs
+   */
+
   organization: Organization;
   project: Project;
   config: Config;
+  /**
+   * List of required feature tags. Note we do not enforce uniqueness of tags anywhere.
+   * On the backend end, feature tags have a scope prefix string that is stripped out on the
+   * frontend (since feature tags are attached to a context object).
+   *
+   * Use `organizations:` or `projects:` prefix strings to specify a feature with context.
+   */
   features: string[];
+  /**
+   * Should the component require all features or just one or more.
+   */
   requireAll?: boolean;
-  renderDisabled?: Function | boolean;
+  /**
+   * Custom renderer function for when the feature is not enabled.
+   *
+   *  - [default] Set this to false to disable rendering anything. If the
+   *    feature is not enabled no children will be rendererd.
+   *
+   *  - Set this to `true` to use the default `ComingSoon` alert component.
+   *
+   *  - Provide a custom render function to customize the rendered component.
+   *
+   * When a custom render function is used, the same object that would be
+   * passed to `children` if a func is provided there, will be used here,
+   * aditionally `children` will also be passed.
+   */
+  renderDisabled?:
+    | ((props: FeatureRenderProps & Pick<Props, 'children'>) => React.ReactNode)
+    | boolean;
+  /**
+   * Specify the key to use for hookstore functionality.
+   *
+   * The hookName should be prefixed with `feature-disabled`.
+   *
+   * When specified, the hookstore will be checked if the feature is
+   * disabled, and the first available hook will be used as the render
+   * function.
+   */
   hookName?: keyof FeatureDisabledHooks;
-  children: React.ReactNode;
+  /**
+   * If children is a function then will be treated as a render prop and
+   * passed FeatureRenderProps.
+   *
+   * The other interface is more simple, only show `children` if org/project has
+   * all the required feature.
+   */
+  children: React.ReactNode | ChildrenRenderFn;
+};
+
+type ChildrenRenderFn = (
+  props: FeatureRenderProps & Pick<Props, 'renderDisabled'>
+) => React.ReactNode;
+
+type FeatureRenderProps = {
+  organization: Organization;
+  project: Project;
+  features: string[];
+  hasFeature: boolean;
+};
+
+type AllFeatures = {
+  configFeatures: string[];
+  organization: string[];
+  project: string[];
 };
 
+function childrenIsRenderProp(children: Props['children']): children is ChildrenRenderFn {
```

Comment:
You could use `isRenderFunc<T>` for this.

Reply:
That's great

Reply:
Adding to diff PR